cmd/storage-datatypes.go

```diff
@@ -296,6 +296,9 @@ func (fi FileInfo) GetDataDir() string {
 // InlineData returns true if object contents are inlined alongside its metadata.
 func (fi FileInfo) InlineData() bool {
 	_, ok := fi.Metadata[ReservedMetadataPrefixLower+"inline-data"]
+	if ok {
+		return !fi.IsRemote()
+	}
 	return ok
```

Comment:
```suggestion
	return ok && !fi.IsRemote()
```

🍰

Reply:
I was trying to avoid the extra checks for non-inlined objects.

Reply:
> I was trying to avoid the extra checks for non-inlined objects.

fi.IsRemove() will not be called if 'ok' is true

Reply:
Can you add a comment to say older versions of MinIO can have inline tag while the object is remote, so it will be easier to understand later

Reply:
> fi.IsRemove() will not be called if 'ok' is true
Yes, that's the intention. In my change, the code checks if an object is tiered only if it has metadata headers suggesting that it was inlined.

Reply:
IsRemote is only called if `ok` is true @krisis it serves your purpose.

Reply:
> IsRemote is only called if ok is true @krisis it serves your purpose.

yes, I had a typo @krisis, I fixed it